File: src/gocode/util/cookie_utilities.go
package util

import (
	"net/http"
	"strings"
)

func GetCookieValue(r *http.Request, cookieName, defaultValue string) string {
	cookies := r.Cookies()
	if len(cookies) > 0 {
		for i := 0; i < len(cookies); i++ {
			cookie := cookies[i]
			if strings.Compare(cookieName, cookie.Name) == 0 {
				return cookie.Value
			}
		}
	}
	return defaultValue
}

func GetCookie(r *http.Request, cookieName string) *http.Cookie {
	cookies := r.Cookies()
	if len(cookies) > 0 {
		for i := 0; i < len(cookies); i++ {
			cookie := cookies[i]
			if strings.Compare(cookieName, cookie.Name) == 0 {
				return cookie
			}
		}
	}
	return nil
}

File: src/util/CookieUtilities.java
package util;

import javax.servlet.http.*;

public class CookieUtilities {
	public static String getCookieValue(HttpServletRequest req, 
			String cookieName, String defaultValue) {
		Cookie[] cookies = req.getCookies();
		if (cookies != null) {
			for (int i = 0; i < cookies.length; i++) {
				Cookie cookie = cookies[i];
				if (cookieName.equals(cookie.getName())) {
					return cookie.getValue();
				}
			}
		}
		return defaultValue;
	}
	
	public static Cookie getCookie(HttpServletRequest req, String cookieName) {
		Cookie[] cookies = req.getCookies();
		if (cookies != null) {
			for (int i = 0; i < cookies.length; i++) {
				Cookie cookie = cookies[i];
				if (cookieName.equals(cookie.getName())) {
					return cookie;
				}
			}
		}
		return null;
	}
}
